### ISSUE TITLE
serialize the models better

### ISSUE THREAD

**Issue by CamDavidsonPilon** [CamDavidsonPilon/lifetimes#97]
https://twitter.com/elesinOlalekan/status/853112408273031169

**Comment by aprotopopov** [comment]
I think [`dill`](https://github.com/uqfoundation/dill) is the way to go, [stackoverflow example and explanation](https://stackoverflow.com/a/25119089/4542359). 
```
import dill
from lifetimes import BetaGeoFitter
from lifetimes.datasets import load_cdnow_summary
cdnow = load_cdnow_summary()
bgf = BetaGeoFitter(penalizer_coef=0.0)
bgf.fit(cdnow['frequency'], cdnow['recency'], cdnow['T'])

with open('bgf.pkl', 'wb') as output:
    dill.dump(bgf, output)
    
with open('bgf.pkl', 'rb') as output:
    bgf_new = dill.load(output)
    
    
print(bgf_new.__dict__['penalizer_coef']             ==  bgf.__dict__['penalizer_coef'])
print(bgf_new.__dict__['_scale']                     ==  bgf.__dict__['_scale'])
print(bgf_new.__dict__['params_']                    ==  bgf.__dict__['params_'])
print(bgf_new.__dict__['_negative_log_likelihood_']  ==  bgf.__dict__['_negative_log_likelihood_'])
print((bgf_new.__dict__['data']                      ==  bgf.__dict__['data']).all().all())
print(bgf_new.__dict__['predict'](1, 1, 2, 5)        ==  bgf.__dict__['predict'](1, 1, 2, 5))

print('generate_date hard to compare')

"""
True
True
True
True
True
True
generate_date hard to compare
"""
```

**Comment by opendining** [comment]
Agreed, we're using dill in production and it's working fine.

Tim Ridgely

**Comment by aprotopopov** [comment]
I think this issue could be closed with [PR#104](https://github.com/CamDavidsonPilon/lifetimes/pull/104)

**Comment by aprotopopov** [comment]
Probably `self.__dict__.update(dill.load(in_file).__dict__)` instead of `self = dill.load(in_file)` will load model with all parameters. [example](https://stackoverflow.com/a/25128214/4542359)
